pkg/cmd/audit/audit_filter.go

```diff
@@ -65,6 +65,28 @@ func (f *FilterByNamespaces) FilterEvents(events ...*auditv1.Event) []*auditv1.E
 	return ret
 }
 
+type FilterBySubresources struct {
+	Subresources sets.String
+}
+
+func (f *FilterBySubresources) FilterEvents(events ...*auditv1.Event) []*auditv1.Event {
+	ret := []*auditv1.Event{}
+	for i := range events {
+		event := events[i]
+		_, _, _, subresource := URIToParts(event.RequestURI)
+
+		if f.Subresources.Has("-*") && len(f.Subresources) == 1 && len(subresource) == 0 {
```

Comment:
`--subresource=-*` will include all events including ones with empty subresource, is that desired? `util.AcceptString` will accept any nonempty subresource if `Subresources == {"-*"}`

Reply:
my intent is that if I specify `-*` and only that, I will get only those results that have no subresource.  I don't think this line includes cases where subresource is greater than zero.

Reply:
yeah, it makes sense. I was thinking of the result set it would produce if a user provided a conflicting requirement `--subresource=-* --subresource=foo`.